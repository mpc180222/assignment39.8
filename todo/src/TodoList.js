import React, { useState } from 'react';
import Todo from './Todo';
import {v4 as uuid} from "uuid";
import NewTodoForm from './NewTodoForm';

const TodoList = () => {

    const INITIAL_STATE = [ 
        {todo: 'Mow lawn',
        id: uuid()},
        {todo: 'Wash dishes',
        id: uuid()}
    ]

    const [todoData, setTodoData] = useState(INITIAL_STATE);

    const addTodo = (newTodo) => {setTodoData(todoData => [...todoData, {...newTodo, id: uuid()}] )}
    const removeTodo = (e) => {
        console.log(e.target.id)
        setTodoData(todoData.filter(item => item.id !== e.target.id))

    }

    return(
        <div>
            {todoData.map(item => <Todo todo = {item.todo} id = {item.id} removeTodo = {removeTodo} />)}
            <NewTodoForm addTodo={addTodo} />
        </div>


    )


}


export default TodoList;