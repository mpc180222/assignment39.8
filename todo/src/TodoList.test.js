import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

it('renders without crashing', () => {
  render(<TodoList />);
  
});

it("should match snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();

})

it("should remove one of the two initial todos", () => {
    const {queryByTestId} = render(<TodoList/>)
    const removeBtn = queryByTestId("remove-Wash dishes");
    fireEvent.click(removeBtn);
    expect(queryByTestId("remove-Wash dishes")).not.toBeInTheDocument();
  })

  it("should add a new todo", () => {
    const {queryByTestId, getByText} = render(<TodoList/>)
    const submitBtn = queryByTestId("submit-todo-input");
    const todoInput = queryByTestId("new-todo-input");
    fireEvent.change(todoInput, {target: {value: "diapers"}});
    fireEvent.click(submitBtn);
    expect(getByText("diapers")).toBeInTheDocument();
  })