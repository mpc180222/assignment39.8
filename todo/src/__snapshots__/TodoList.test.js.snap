// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h2>
        Mow lawn
      </h2>
      <button
        data-testid="remove-Mow lawn"
        id="fa9c0637-b929-4ee9-9a30-b84a584e5bf8"
      >
        X
      </button>
    </div>
    <div>
      <h2>
        Wash dishes
      </h2>
      <button
        data-testid="remove-Wash dishes"
        id="8078b0f9-7ba2-4e50-8fff-5e420fc02780"
      >
        X
      </button>
    </div>
    <form>
      <input
        data-testid="new-todo-input"
        name="todo"
        placeholder="new todo"
        value=""
      />
      <button
        data-testid="submit-todo-input"
      >
        Add todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
